### ISSUE TITLE
Fold forge snapshot into forge test

### ISSUE THREAD

**Issue by CodeSandwich** [foundry-rs/foundry#7201]
### Component

Forge

### Describe the feature you would like

The existence of the `forge snapshot` command seems excessive, maybe it should be folded into `forge test` by adding the `--snap` `--diff`, `--check` and `--tolerance` options? 

#### Rationale
`forge snapshot` can't be used for anything other than tests, it's an exact copy of `forge test` with a few extra options. It's confusing to use because it's a complicated command like `forge test` but despite seeming to be a new, difficult to master tool it's identical. It's hard to find because usually people will be using `forge test` to benchmark and comparing gas snapshots feels like a feature of that command, not a separate operations.

**Comment by grandizzy** [comment]
@CodeSandwich we added cheatcodes to address similar, pls check and use https://book.getfoundry.sh/forge/gas-section-snapshots
Thank you!

**Comment by CodeSandwich** [comment]
That's a really neat set of tools! It has a different philosophy from `forge snapshot` by measuring manually chosen bits and not whole test runs, and IMO it's a far more useful approach. Is `forge snapshot` deprecated now? If it isn't, the issue is still unresolved because the CLI still has a huge redundant command.

**Comment by grandizzy** [comment]
not deprecated but could be something to do. @zerosnacks can you pls chime in?

**Comment by zerosnacks** [comment]
`forge snapshot` is still used quite a bit and is still a reasonable way to get a quick overview as it is much less invasive than cheatcodes inside of tests. I think both can continue to co-exist, at least for the foreseeable future.

As an example, Solady uses https://github.com/atarpara/foundry-snapshot-diff

**Comment by grandizzy** [comment]
makes sense, so we keep and not going to add too many things to it but encouraging people to use the new way
